### ISSUE TITLE
`action(fn)` stop the autorun when throw an Error

### ISSUE THREAD

**Issue by xiamidaxia** [mobxjs/mobx#286]
```javascript

    let autorunTimes = 0;
    class Todos {
      @observable count = 0;
      @action add() {
        this.count ++;
        if (this.count === 2) {
          throw new Error('An Action Error!');
        }
      }
    }
    const todo = new Todos;
    autorun(() => {
      autorunTimes ++;
      return todo.count;
    });
    try {
      todo.add();
      expect(autorunTimes).to.eql(2);
      todo.add();
    } catch (e) {
      todo.add();
      // Expect 3 but get 2
      expect(autorunTimes).to.eql(3);
    }

```

**Comment by AriaFallah** [comment]
Does this happen even without `@action`?

Also not 100% sure on this, but isn't this the expected behavior? If something autorun is observing throws an error, shouldn't it stop observing it?

**Comment by xiamidaxia** [comment]
All the autorun will be invalid when the `@action` throw an error , So this will make affect on global data.

**Comment by mweststrate** [comment]
Will be fixed in 2.3
